constraint/pkg/types/validation.go

```diff
@@ -134,3 +136,13 @@ func (r *Responses) TraceDump() string {
 	}
 	return b.String()
 }
+
+// ResultMeta defines an interface to expose metadata for a Result.
+type ResultMeta interface {
+	// EngineStatsString gives an amorphous representation of engine stats
+	// such as latency, engine type or the number of constraints evalauted
+	// against a template. The underlying enigne type defines and implements
+	// structs for this interface. This function is primarily inteded for logging
+	// out for the consumers of this library.
+	EngineStatsString() string
```

Comment:
no need for `String`

Another option here would be to return map[string]interface{}, which would make it play a bit more nicely with other serialization pipelines (such as the logs pipeline), and allow for programatic inspection should someone want to try.

Reply:
I changed the return type to `map[string]interface{]` but also the signature to `ditto , error`; where an error occurs on json.marshalling/ unmarshalling